packages/backend-data/src/factory.ts

```diff
@@ -171,15 +177,57 @@ class DataGenerator implements ConstructContainerEntryGenerator {
       authorizationModes,
       outputStorageStrategy: this.outputStorageStrategy,
       functionNameMap,
-      translationBehavior: { sandboxModeEnabled },
+      translationBehavior: {
+        sandboxModeEnabled,
+        /**
+         * The destructive updates should be always allowed in backend definition and not to be controlled on the IaC
+         * The CI/CD check should take the responsibility to validate if any tables are being replaced and determine whether to execute the changeset
+         */
+        allowDestructiveGraphqlSchemaUpdates: true,
+      },
     });
+    Aspects.of(amplifyApi).add(new ReplaceTableUponGsiUpdateChecker());
 
     convertJsResolverDefinition(scope, amplifyApi, jsFunctions);
 
     return amplifyApi;
   };
 }
 
+const REPLACE_TABLE_UPON_GSI_UPDATE_ATTRIBUTE_NAME: keyof TranslationBehavior =
+  'replaceTableUponGsiUpdate';
+
+enum BackendDataEnvironmentVariables {
+  // Environment variables allowed in sandbox mode only
+  REPLACE_TABLE_UPON_GSI_UPDATE = 'REPLACE_TABLE_UPON_GSI_UPDATE',
```

Comment:
```suggestion
  AMPLIFY_BACKEND_DATA_REPLACE_TABLE_UPON_GSI_UPDATE = 'AMPLIFY_BACKEND_DATA_REPLACE_TABLE_UPON_GSI_UPDATE',
```

Reply:
The env var is removed as it is always allowed in sandbox